apps/sms/services/test/unit/conversation/conversation_service_test.js

```diff
@@ -0,0 +1,318 @@
+/*global bridge,
+         BroadcastChannel,
+         ConversationService,
+         Drafts,
+         MocksHelper
+*/
+
+'use strict';
+
+require('/services/test/unit/mock_bridge.js');
+require('/services/test/unit/mock_drafts.js');
+require('/services/js/bridge_service_mixin.js');
+require('/views/shared/test/unit/mock_broadcast_channel.js');
+require('/services/js/conversation/conversation_service.js');
+
+var MocksHelperForAttachment = new MocksHelper([
+  'bridge',
+  'BroadcastChannel',
+  'Drafts',
+  'Draft',
+  'streamClient',
+  'streamService'
+]).init();
+
+suite('ConversationService >', function() {
+  var serviceStub, mobileMessageClientStub, mobileMessageStreamStub;
+
+  MocksHelperForAttachment.attachTestHelpers();
+
+  setup(function() {
+    serviceStub = sinon.stub({
```

Comment:
I admit this test file is not easy to read with all that stubs, but our case is not that easy as well :) So if you see any good ways to make it more readable - I'm all ears!